modules/core/include/opencv2/core/dualquaternion.inl.hpp

```diff
@@ -158,7 +158,7 @@ inline Quat<T> DualQuat<T>::getRotation(QuatAssumeType assumeUnit) const
 template <typename T>
 inline Vec<T, 3> DualQuat<T>::getTranslation(QuatAssumeType assumeUnit) const
 {
-    Quat<T> trans = 2.0 * (getDualPart() * getRealPart().inv(assumeUnit));
+    Quat<T> trans = T(2.0) * (getDualPart() * getRealPart().inv(assumeUnit));
```

Comment:
`2.0f` ?

Reply:
`T` type can be float or double, so what's the point of using `f` postfix?

Reply:
f suffix makes number of float type.
AFAIK, there is no issues with both operations `float*float` or `float*double` (the second result is double).

Reply:
It's understood; the problem is that this is not `float*float` or `float*double`, this is `float*Quat<T>` where `T` can be both.
No wonder using 2.0f breaks compilation on my machine (fails to find corresponding operator).

I propose to leave this code as it is now.